src/tree/mod.rs

```diff
@@ -639,8 +649,8 @@ impl RatchetTree {
         &mut self,
         proposal_queue: ProposalQueue,
         updates_key_package_bundles: &[KeyPackageBundle],
-        // (path_required, self_removed, invitation_list)
-    ) -> Result<(bool, bool, InvitationList), TreeError> {
+        // (path_required, self_removed, invitation_list, new_leaves_indexes)
+    ) -> Result<(bool, bool, InvitationList, HashSet<NodeIndex>), TreeError> {
```

Comment:
These return values get pretty wild... at some point we should introduce structs to hold these complex values.